src/utils/url/getAbsoluteUrl.ts

```diff
@@ -11,5 +11,5 @@ export function getAbsoluteUrl(path: string, baseUrl?: string): string {
   // or the current location (in the case of browser/browser-like environments).
   const origin = baseUrl || (typeof location !== 'undefined' && location.origin)
 
-  return origin ? new URL(path, origin).href : path
+  return origin ? decodeURI(new URL(encodeURI(path), origin).href) : path
```

Comment:
Encoding and decoding the URL preserves the escaped characters present in the `path`.